### PR TITLE
adjusted modify_rotate to preserve nonlinear scale data

### PR DESCRIPTION
**Pull request by vsicurella**
Fix for #140 

Instead of transposing in-place, we start transposing on the line after the new root through the equave. The transposed equave becomes the degree by which you can use to invert the rotation. Then we cycle back to the beginning of the scale and transpose up to the new root. Then we push the original equave to the end of the scale.

### CODE REVIEW

File: src/js/modifiers.js [hunk 1]
```diff
@@ -275,23 +275,24 @@ function modify_rotate() {
...
+  let rotatedLines = []
+
+  // transpose lines, mod equave
+  // start on degree after transposer/new root, cycle around and stop before transposer
+  for (let i = 1; i < lines.length; i++) {
+    let deg = (i + degree) % lines.length
+    let index = i - 1
+    rotatedLines[index] = moduloLine(transposeLine(lines[deg], transposer), equave)
+    console.log(index)
     console.log(degree)
```
Comment (meszaros-lajos-gyorgy) [inline src/js/modifiers.js:287]: I would recommend cleaning up `console.log` calls once the development of a feature have been finished.

File: src/js/modifiers.js [hunk 2]
```diff
@@ -275,23 +275,24 @@ function modify_rotate() {
   console.log(transposer)
   console.log(equave)
```
Comment (meszaros-lajos-gyorgy) [inline src/js/modifiers.js:276]: I know the PR isn't about these lines, but if possible and no longer needed, then please remove these console logs

**Review by meszaros-lajos-gyorgy — CHANGES_REQUESTED** [review]
Please check if those console.logs are needed and if not then please remove them to minimize accidental console spamming.
Apart from that I like the renaming of variables and the reorganizing of code, the code seems more readable to me.

**Comment by vsicurella** [comment]
no worries, they were in there previously, but since the sorting issue is resolved I don't think they're needed anymore.

**Review by meszaros-lajos-gyorgy — APPROVED** [review]
Nicely done!

**Pull request merged**